backend/chromeApiServer/src/main/java/com/canvas/service/EvaluationService.java

```diff
@@ -0,0 +1,157 @@
+package com.canvas.service;
+
+import com.canvas.service.helperServices.CanvasClientService;
+import com.canvas.service.helperServices.FileService;
+import com.canvas.service.helperServices.ProcessExecutor;
+import com.canvas.service.models.CommandOutput;
+import com.canvas.service.models.ExtensionUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles the Student side of the evaluation process.
+ */
+@Service
+public class EvaluationService {
+    private final FileService fileService;
+    private final CanvasClientService canvasClientService;
+    String MAKEFILE = "makefile";
+
+    /**
+     * Constructor for creating our instances of FileService and CanvasClientService
+     *
+     * @param fileService         File service instance
+     * @param canvasClientService Canvas client service instance
+     */
+    @Autowired
+    public EvaluationService(FileService fileService, CanvasClientService canvasClientService) {
+        this.fileService = fileService;
+        this.canvasClientService = canvasClientService;
+    }
+
+    /**
+     * This method will compile the student code. The files are extracted
+     * and the program is compiled.
+     *
+     * @param user  User object holding all components associated with a user
+     * @param files The files to compile
+     * @return      return a CommandOutput response
+     */
+    public ResponseEntity<CommandOutput> compileStudentCodeFile(ExtensionUser user, MultipartFile[] files) {
```

Comment:
would it make more sense to return just a `CommandOuput` instead of a `ResponseEntity`, and instead let the controller construct the `ResponseEntity`?

Reply:
I kept it as Repsonse entity as I was mirroring the tutorial for java springboot. It makes it easier to just pass the expected response to the API so there's less for the API to handle.